### ISSUE TITLE
False positives with `Style/FetchEnvVar` in conditions

### ISSUE THREAD

**Issue by inkstak** [rubocop/rubocop#10652]
## Actual behavior

@koic added in #10603:

> The ENV var as a if condition is currently allowed.
> So the ENV var already used in the condition will be accepted as well.

````ruby
if ENV["SOME_VAR"]
  do_something ENV["SOME_VAR"]
end
````

Nothing to say about this snippet.  
But it won't work with more complex conditions:

````ruby
if some_condition || ENV["FOO"] 
  do_something ENV.fetch("FOO")
end

if ENV["BAR"] || some_condition
  do_something ENV.fetch("BAR")
end

# Style/FetchEnvVar: Use ENV.fetch("FOO") or ENV.fetch("FOO", nil) instead of ENV["FOO"].
#   if some_condition || ENV["FOO"]
#                        ^^^^^^^^^^^
# Style/FetchEnvVar: Use ENV.fetch("BAR") { some_condition } instead of ENV["BAR"] || some_condition.
#   if ENV["BAR"] || some_condition
#      ^^^^^^^^^^^
````

I don't think the cop should suggest to use `fetch` in the first condition `if some_condition || ENV["FOO"]` just as `if ENV["FOO"]` is allowed.

The second condition might be subjective: `ENV.fetch("BAR") { some_condition }` is correct but just because the first example should not report any offenses, this one should not suggest to use `fetch` in conditions. 

There no issue with the actual behavior inside the `if` blocks:
as `some_condition` might be true, we should use `do_something ENV.fetch("VAR1")`

````ruby
if ENV["ITEM_1"] == 1
  do_something ENV["ITEM_1"]
end

if ENV["ITEM_2"].present?
  do_something ENV["ITEM_2"]
end

if ENV["ITEM_3"].in?(%w[A B C])
  do_something_else ENV["ITEM_3"]
end

if ENV.key?("ITEM_4")
  do_something ENV["ITEM_4"]
end

# Style/FetchEnvVar: Use ENV.fetch("ITEM_1") or ENV.fetch("ITEM_1", nil) instead of ENV["ITEM_1"].
#     do_something ENV["ITEM_1"]
#                  ^^^^^^^^^^^^^
# Style/FetchEnvVar: Use ENV.fetch("ITEM_2") or ENV.fetch("ITEM_2", nil) instead of ENV["ITEM_2"].
#     do_something ENV["ITEM_2"]
#                  ^^^^^^^^^^^^^
# Style/FetchEnvVar: Use ENV.fetch("ITEM_3") or ENV.fetch("ITEM_3", nil) instead of ENV["ITEM_3"].
#     do_something ENV["ITEM_3"]
#                  ^^^^^^^^^^^^^
# Style/FetchEnvVar: Use ENV.fetch("ITEM_4") or ENV.fetch("ITEM_3", nil) instead of ENV["ITEM_3"].
#     do_something ENV["ITEM_4"]
#                  ^^^^^^^^^^^^^
````

Is it possible to accept these situation as the ENV var has been "tested" in conditions ?
I guess it is complicated to tell if the methods used in conditions are testing or not the variable.
Allowing equallity and predicate methods could be a start ?

````ruby
if %w[A B C].include?(ENV["OPTION"])
  do_something ENV["OPTION"]
end

# Style/FetchEnvVar: Use ENV.fetch("OPTION") or ENV.fetch("OPTION", nil) instead of ENV["OPTION"].
#   if %w[A B C].include?(ENV["OPTION"])
#                         ^^^^^^^^^^^^^
# Style/FetchEnvVar: Use ENV.fetch("OPTION") or ENV.fetch("OPTION", nil) instead of ENV["OPTION"].
#     do_something ENV["OPTION"]
#                  ^^^^^^^^^^^^^
````

This last one might be more complicated.
Is it possible to catch and exclude this situation ?
There is no point in using fetch in the `if` condition nor inside the `id` block.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.29.1 (using Parser 3.1.2.0, rubocop-ast 1.18.0, running on ruby 2.7.6 x86_64-darwin21)
  - rubocop-performance 1.13.3
  - rubocop-rails 2.14.2
  - rubocop-rspec 2.11.1
```
